Issue 161: by column with plot_grid

BrianLang opened the pull request:
closes #161
- merged to master 
- rebuilt documentation 
- added visual test 
- validated visual test

BrianLang commented:
Somehow I'm still figuring out how to do all of this. Was there a way to do what you asked (merge to master, fix things, commit, and push) without opening a new pull request?

clauswilke commented:
Yes, you can just keep pushing commits and github automatically sees them and updates the PR. No problem though, we can work with this one.

clauswilke commented:
Sorry, one more issue. Can you try your plot reordering code for a case where `num_plots` is less than `rows * cols`? I think your `matrix()` call will create a warning, and you should be able to fix this by padding `1:num_plots` with `NA`s. Maybe this will work: `c(1:num_plots, rep(NA, rows * cols - num_plots))`

clauswilke commented on tests/testthat/test_plot_grid.R:
```diff
@@ -27,6 +27,10 @@ test_that("basic plot arranging works", {
   )
+
+  expect_doppelganger("colwise arranging",
+    plot_grid(p1, NULL, p2, NULL, p3, NULL, byrow = FALSE) + theme_map()
```
I'd delete the last `NULL` argument to test the case that the number of plots is less than `rows * cols`.

BrianLang replied:
Thanks, yes, of course this could break things. I'll check and fix it: adding a visual test that assesses the case where the number of plots is less than `rows * cols`

BrianLang commented:
You were correct that having `num_plots < cols * rows` produces warnings and issues, specifically that matrix recycles inputs. So I've added your suggestion of padding with `c(1:num_plots, rep(NA, rows * cols - num_plots))` and it looks correct now.

It also is coincidental the case that the previous visual test gave a false "valid" for dropping the last NULL, so I've updated it to something that wouldn't pass given the previous implementation.
> ```
>expect_doppelganger("colwise arranging",
>    plot_grid(p1, NULL, p2, p3, NULL, byrow = FALSE) + theme_map() # add theme_map() for plot title
>  )
>```

clauswilke commented:
Thanks!

BrianLang commented:
Woooo! First merged pull request! Thanks for your help :-)
